flocker/node/agents/functional/test_cinder.py

```diff
@@ -39,9 +39,20 @@ def cinderblockdeviceapi_for_test(test_case, cluster_id):
         by ``TidyCinderVolumeManager`` to cleanup any lingering volumes that
         are created during the course of ``test_case``
     """
+    local_instance_uuid = _instance_uuid()
+    # Create the nova client last so that its cleanup method is
+    # run first and detaches volumes before the cinder client
+    # cleanup then deletes them.
+    # Yuck!
```

Comment:
I should probably combine these into a single test helper that calls the cleanup operations in the correct order.